music21/midi/translate.py

```diff
@@ -4016,6 +4057,48 @@ def testImportInstrumentsWithoutProgramChanges(self):
         meta_event_pairs = getMetaEvents([(DUMMY_DELTA_TIME, event2)])
         self.assertEqual(meta_event_pairs[0][1].midiProgram, 53)
 
+    def testExportUnpitched(self):
+        from music21 import midi as midiModule
+
+        m = stream.Measure(
+            [instrument.BassDrum(),
+            note.Unpitched(),
+            percussion.PercussionChord([note.Unpitched(), note.Unpitched()])]
+        )
+        trks = streamHierarchyToMidiTracks(m)
+        bd_trk = trks[1]
+
+        self.assertTrue({ev.channel for ev in bd_trk.events}, {10})
+        note_ons = [
+            ev for ev in bd_trk.events if ev.type is midiModule.ChannelVoiceMessages.NOTE_ON]
+        self.assertEqual(len(note_ons), 3)
+        self.assertEqual({ev.pitch for ev in note_ons}, {35})
+
+        # Replace the BassDrum with a vague instrument (lossy import)
+        m.pop(0)
+        m.insert(0, instrument.Instrument())
+
+        trks = streamHierarchyToMidiTracks(m)
+        drum_trk = trks[1]
+
+        self.assertTrue({ev.channel for ev in drum_trk.events}, {10})
+        note_ons = [
+            ev for ev in drum_trk.events if ev.type is midiModule.ChannelVoiceMessages.NOTE_ON]
+        self.assertEqual(len(note_ons), 3)
+        self.assertEqual({ev.pitch for ev in note_ons}, {60})  # fallback
+
+        # Change the stored instrument: affects that note only
+        m.notes.first().storedInstrument = instrument.Agogo()
+
+        trks = streamHierarchyToMidiTracks(m)
+        mixed_trk = trks[1]
+
+        self.assertTrue({ev.channel for ev in mixed_trk.events}, {10})
+        note_ons = [
+            ev for ev in mixed_trk.events if ev.type is midiModule.ChannelVoiceMessages.NOTE_ON]
+        self.assertEqual(len(note_ons), 3)
+        self.assertEqual([ev.pitch for ev in note_ons], [67, 60, 60])
```

Comment:
Yep, if I'm following, then that's tested here: put Agogo on the first unpitched note in the stream and get midi note 67 on that one only.